test/DebugInfo/DebugInfoUnsupported.ll

```diff
@@ -0,0 +1,42 @@
+; This test checks that metadata with unknown DWARF operations is translated
+; to "empty" debug info to SPIR-V and back.
+; TODO: remove this test as soon as more DWARF operations are supported.
+
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv %t.spv -to-text -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV: DebugInfoNone
+; CHECK-LLVM-NOT: DW_OP_LLVM_arg
+
+target datalayout = "e-m:o-i64:64-f80:128-n8:16:32:64-S128"
+target triple = "spir64-unknown-unknown"
+
+declare void @llvm.dbg.value(metadata, metadata, metadata) nounwind readnone speculatable
+
+define void @DbgIntrinsics() sanitize_memtag {
+entry:
+  %x = alloca i32, align 4
+  call void @llvm.dbg.value(metadata !DIArgList(i32* %x, i32* %x), metadata !6, metadata !DIExpression(DW_OP_LLVM_arg, 0, DW_OP_LLVM_arg, 1, DW_OP_plus)), !dbg !10
```

Comment:
Updated in 715e6bb

Reply:
What happened with DIArgList? It seems right, but I don't recall if we added handling for this type of metadata.

Reply:
You're right, there is no special handling for this metadata.
The translator just takes the variable (from the second intrinsic argument) instead of handling the first operand of intrinsic which actually contains the `DIArgList`:
https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/1cad18c0c448650d1cc3f18f6077360a98a752f9/lib/SPIRV/LLVMToSPIRVDbgTran.cpp#L120-L121
So, this behavior should be changed once we start to support DIArgList, but it's not related to the changes in this PR.